Common/Data/Consolidators/PsychSignalConsolidator.cs

```diff
@@ -0,0 +1,114 @@
+        /// </summary>
+        /// <param name="maxCount">The number of pieces to accept before emiting <see cref="PsychSignalConsolidated"/></param>
+        /// <param name="period">The minimum span of time before emitting <see cref="PsychSignalConsolidated"/></param>
+        public PsychSignalConsolidator(int maxCount, TimeSpan period)
+            : base(maxCount, period)
+        {
+        }
+
+        /// <summary>
+        /// Creates a consolidator to produce a new 'PsychSignalConsolidated' representing the last count pieces of data or the period, whichever comes first
+        /// </summary>
+        /// <param name="func">Func that defines the start time of a consolidated data</param>
+        public PsychSignalConsolidator(Func<DateTime, CalendarInfo> func)
+            : base(func)
+        {
+        }
+
+        /// <summary>
+        /// Aggregates the new 'data' into the 'workingBar'. The 'workingBar' will be
+        /// null following the event firing
+        /// </summary>
+        /// <param name="workingBar">The <see cref="PsychSignalConsolidated"/> instance we're building, null if the event was just fired and we're starting a new instance</param>
+        /// <param name="data">The new data</param>
+        protected override void AggregateBar(ref PsychSignalConsolidated workingBar, PsychSignalSentimentData data)
+        {
+            if (workingBar == null)
+            {
+                workingBar = new PsychSignalConsolidated
+                {
+                    Symbol = data.Symbol,
+                    Time = GetRoundedBarTime(data.Time),
+                    BullIntensity = new Bar(data.BullIntensity, data.BullIntensity, data.BullIntensity, data.BullIntensity),
+                    BearIntensity = new Bar(data.BearIntensity, data.BearIntensity, data.BearIntensity, data.BearIntensity),
+                    BullMinusBear = new Bar(data.BullMinusBear, data.BullMinusBear, data.BullMinusBear, data.BullMinusBear),
+                    BullScoredMessages = data.BullScoredMessages,
+                    BearScoredMessages = data.BearScoredMessages,
+                    BullBearMessageRatio = new Bar(data.BullBearMessageRatio, data.BullBearMessageRatio, data.BullBearMessageRatio, data.BullBearMessageRatio),
+                    TotalScoredMessages = data.TotalScoredMessages
+                };
+
+                return;
+            }
+
+            UpdateBar(workingBar.BullIntensity, data.BullIntensity);
+            UpdateBar(workingBar.BearIntensity, data.BearIntensity);
+            UpdateBar(workingBar.BullMinusBear, data.BullMinusBear);
+            workingBar.BullScoredMessages += data.BullScoredMessages;
+            workingBar.BearScoredMessages += data.BearScoredMessages;
+            UpdateBar(workingBar.BullBearMessageRatio, data.BullBearMessageRatio);
+            workingBar.TotalScoredMessages += data.TotalScoredMessages;
+            workingBar.EndTime = GetRoundedBarTime(data.EndTime);
```

Comment:
just checking: are `PsychSignalSentimentData` members always set in each data point? peaked on some of them and saw a bunch of `0` - if not, thinking if those cases should be skipped/ignored when consolidating? Can follow up offline

Reply:
PsychSignal data can contain a lot of `0` values, but will always contain at least one value per row of data. This means that the consolidator will always be doing something, and not be consolidating empty data.

Reply:
Hm given it can contain a lot of 0 values, I'm trying to think of the use case here, user perspective, consider first we get `data.BullIntensity` `1` and afterward we get a bunch of `0s` -> what does the consolidated data, as a `Bar` instance, mean? What's the value in it? 🤔

Reply:
Data provider specified that 0 values could be a neutral score (information) or a no score/message case (not information, should be skipped), and that we should `differentiate them checking the bull_scored_messages and bear_scored_messages fields to see if the count is "0".`